applications/StructuralMechanicsApplication/python_scripts/structural_mechanics_adjoint_static_solver.py

```diff
@@ -48,7 +48,33 @@ def PrepareModelPart(self):
             raise Exception("there are currently only 3D adjoint elements available")
         super(StructuralMechanicsAdjointStaticSolver, self).PrepareModelPart()
         # TODO Why does replacement need to happen after reading materials?
-        StructuralMechanicsApplication.ReplaceElementsAndConditionsForAdjointProblemProcess(self.main_model_part).Execute()
+
+        process_info = self.main_model_part.ProcessInfo
+        if (process_info.Has(StructuralMechanicsApplication.IS_ADJOINT) and
+            process_info.GetValue(StructuralMechanicsApplication.IS_ADJOINT)):
+            raise RuntimeError("Modelpart '{}' is already adjoint modelpart!".format(self.main_model_part.Name))
+
+        # defines how the primal elements should be replaced with their adjoint counterparts
+        replacement_settings = KratosMultiphysics.Parameters("""
+            {
+                "element_name_table" :
+                {
+                    "ShellThinElement3D3N"           : "AdjointFiniteDifferencingShellThinElement3D3N",
+                    "CrLinearBeamElement3D2N"        : "AdjointFiniteDifferenceCrBeamElementLinear3D2N",
+                    "TrussLinearElement3D2N"         : "AdjointFiniteDifferenceTrussLinearElement3D2N",
+                    "TrussElement3D2N"               : "AdjointFiniteDifferenceTrussElement3D2N"
+                },
+                "condition_name_table" :
+                {
+                    "PointLoadCondition2D1N"         : "AdjointSemiAnalyticPointLoadCondition2D1N",
+                    "PointLoadCondition3D1N"         : "AdjointSemiAnalyticPointLoadCondition3D1N"
```

Comment:
Just wondering how you proceed now with your `ShapeOptimizationCondition`?

Reply:
With the current implementation there are two options:

1. The process has a flag that it ignores if conditions/elements from the model part are not found in the table. like this e.g. the ShapeOptimizationConditions would simply be ignored and not replaced. This has the disadvantage that silently also undefined structural elements would be ignored...
2. One could also define that an element/condition should be replaced with the same type by adding e.g. `"ShapeOptimizationCondition3D3N" : "ShapeOptimizationCondition3D3N"` to the settings. The advantage is that nothing will be ignored silently. The disadvantage is that this only works if the ShapeOptimizationCondition is registered (happens only if ShapeOptimizationApplication is imported...)

I see an alternative to extend the settings of the replacement process to explicitly define element/conditions that should be ignored. This would probably the cleanest option. 
```python
"ignore_elements" : [
    "element1",
    "element2"
],
"ignore_conditions" : [
   ...
]
```

Reply:
@armingeiser side question, I though you no longer need the ShapeOptCondition?

Reply:
Potentially we can replace them with the "default" conditions from the core, but I did not try yet.

Reply:
On your questions: please check #4387

Reply:
I would go for a list of elems/conds to be ignored
Then the user can customize it to any needs

Reply:
Do you think the `throw_error` flag is still necessary in that case? That's kind of a wildcard to ignore everything that isn't defined in the table.

Reply:
I would say this can be necessary
ofc the default is to throw